File: src/main/java/com/lkimilhol/healingwords/words/domain/Writer.java
package com.lkimilhol.healingwords.words.domain;

import javax.persistence.Column;
import javax.persistence.Embeddable;

import lombok.NoArgsConstructor;

import com.lkimilhol.healingwords.exception.WriterLengthExceedException;

@Embeddable
@NoArgsConstructor
public class Writer {
    public static final int MAX_LENGTH = 15;

    @Column(name = "writer", nullable = false)
    private String name;

    public Writer(String name) {
        checkLength(name);
        this.name = name;
    }

    private void checkLength(String name) {
        if (name.length() > MAX_LENGTH) {
            throw new WriterLengthExceedException();
        }
    }
}

File: src/main/kotlin/com/lkimilhol/healingwords/words/domain/Writer.kt
package com.lkimilhol.healingwords.words.domain

import com.lkimilhol.healingwords.exception.WriterLengthExceedException
import javax.persistence.Column
import javax.persistence.Embeddable

@Embeddable
class Writer private constructor(
    @Column(name = "writer", nullable = false)
    private val name: String
) {

    fun name(): String {
        return name
    }

    companion object {
        const val MAX_LENGTH = 15

        fun create(name: String): Writer {
            checkLength(name)
            return Writer(name)
        }

        private fun checkLength(name: String) {
            if (name.length > MAX_LENGTH) {
                throw WriterLengthExceedException()
            }
        }
    }
}